src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -210,19 +210,15 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
             );
     }
 
-    public boolean hasIdenticalMappings(TermFieldScope scope, Map<String, String> indexToType) {
```

Comment:
I realized that relaxing the mapping check completely will break the functionality where mappings are used to generate Elasticsearch DSL such as `WHERE =` clause, `GROUP BY`, `ORDER By ` etc. This is because it will make it ambiguous as to mapping from which index to be applied. Lets  keep this function, but only check for mappings present as part of `SELECT` clause.